backend/gamification/gamification.go

```diff
@@ -0,0 +1,94 @@
+// Package gamification handles the gamification within the application, to reward users for performing security checks and staying secure.
+//
+// Exported function(s): PointCalculation, LighthouseStateTransition
+package gamification
+
+import (
+	"database/sql"
+	"strconv"
+
+	"github.com/InfoSec-Agent/InfoSec-Agent/backend/checks"
+	"github.com/InfoSec-Agent/InfoSec-Agent/backend/database"
+	"github.com/InfoSec-Agent/InfoSec-Agent/backend/logger"
+)
+
+// GameState is a struct that represents the state of the gamification.
+// This consists of the user's points, a history of all previous points, and a lighthouse state.
+type GameState struct {
+	Points          int
+	PointsHistory   []int
+	LighthouseState int
+}
+
+// PointCalculation calculates the number of points for the user based on the check results.
+//
+// Parameters:
+//   - gs (GameState): The current game state, which includes the user's points and lighthouse state.
+//   - securityChecks ([] func() checks.Check): A slice of check functions, which will be executed to calculate the points amount.
+//
+// TO DO: points based on more factors than only the checks.
+//
+// Returns:
+//   - GameState: The updated game state with the new points amount.
+func PointCalculation(gs GameState, securityChecks []func() checks.Check) (GameState, error) {
```

Comment:
Je geeft nu de lijst met functies mee, waardoor je als het ware opnieuw een scan uitvoert. Ik denk dat het beter is als je als parameter de lijst van checkresult objecten hebt, met de resultaten van een scan. Dan voer je hier niet nog een keer alle checks uit, en dan kan je ook makkelijker testcases maken, van speciefieke scenarios binnen checkresults. Coverage is van wat ik nu zie in golang rond de 40%, dit kan dan makkelijker omhoog door specifieke resultaten mee te geven.